src/setup/node.rs

```diff
@@ -26,25 +36,48 @@ pub struct Node {
 }
 
 impl Node {
+    /// Creates a new running node with the config and database loaded from the predefined state directory.
+    /// Before the start, content is copied to a new temporary directory and loaded from there.
+    /// This is done to avoid `poisoning` the predefined state directory.
+    pub async fn stateful() -> Result<Node> {
+        let from = build_stateful_path()?;
+        let temp_dir = TempDir::new()?;
+        copy(from, &temp_dir, &CopyOptions::new())?;
+        let mut meta = NodeMetaData::new(temp_dir.path().to_path_buf().join(STATEFUL_DIR))?;
+        meta.start_args.append(&mut vec![
+            "--valid".into(),
+            "--quorum".into(),
+            "1".into(),
+            "--load".into(),
+        ]);
+        let process = process::start(&meta, true);
+        wait_for_start(&SocketAddr::V4(SocketAddrV4::new(
+            STATEFUL_IP,
+            JSON_RPC_PORT,
+        )))
+        .await;
+        Ok(Node {
```

Comment:
Suggestion: leave whitespace before return values to make them quickly parsable.

Reply:
Done